File: ClientApp/angular/src/app/shared/models/order.ts
import { IAddress } from './address';

export interface IOrderToCreate {
    cartId: number;
    deliveryMethodId: number;
    shipToAddress: IAddress;
}

export interface IOrder {
    id: number;
    userEmail: string;
    orderDate: string;
    address: IAddress;
    delivery: string;
    shippingPrice: number;
    orderItems: IOrderItem[];
    subtotal: number;
    total: number;
    status: string;
    paymentIntentId: string;
}

export interface IOrderItem {
    carId: number;
    carName: string;
    pictureUrl: string;
    price: number;
    quantity: number;
}

File: CarsShop/CarsShop.Domain/Order/Order.cs
namespace CarsShop.Domain.Order;

public class Order : BaseEntity
{
    
    public string UserEmail { get; set; }
    public DateTimeOffset OrderDate { get; set; } = DateTimeOffset.Now;
    public IEnumerable<OrderItem> OrderItems { get; set; }
    public decimal Subtotal { get; set; }
        
    public string Status { get; set; } = "Pending";
    
    public Address Address { get; set; }
    public Delivery Delivery { get; set; }

    public decimal Total()
    {
        return Subtotal + Delivery.Price;
    }

    public Order(string userEmail, IEnumerable<OrderItem> orderItems, decimal subtotal,
        Delivery delivery,Address address)
    {
        UserEmail = userEmail;
        OrderItems = orderItems;
        Subtotal = subtotal;
        Delivery = delivery;
        Address = address;
    }

    public Order()
    {
    }
}